pkg/asset/agent/image/baseiso.go

```diff
@@ -0,0 +1,139 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/rhcos"
+)
+
+// BaseIso generates the base ISO file for the image
+type BaseIso struct {
+	File *asset.File
+}
+
+var (
+	baseIsoFilename = ""
+)
+
+var _ asset.WritableAsset = (*BaseIso)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BaseIso) Name() string {
+	return "BaseIso"
+}
+
+// getIsoFile is a pluggable function that gets the base ISO file
+type getIsoFile func() (string, error)
+
+type getIso struct {
+	getter getIsoFile
+}
+
+func newGetIso(getter getIsoFile) *getIso {
+	return &getIso{getter: getter}
+}
+
+// GetIsoPluggable defines the method to use get the baseIso file
+var GetIsoPluggable = downloadIso
+
+// Download the ISO using the URL in rhcos.json
+func downloadIso() (string, error) {
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
```

Comment:
Probably this function could be refactored in future to remove some duplication (and I guess using also the `Image` asset directly once we could add a dependency to `InstallConfig`), but for now it could be reasonable for me

Reply:
Yes once InstallConfig is used we can remove some of this code, will leave it here for now